lib/s3routes/routesUtils.ts

```diff
@@ -106,34 +104,36 @@ const XMLResponseBackend = {
         log.debug('response http code', { httpCode: 200 });
         log.trace('xml response', { xml });
         return response.end(xml, 'utf8', () => {
-            // TODO ARSN-216 Fix logger
-            // @ts-expect-error
-            log.end().info('responded with XML', {
+            log.end('responded with XML', {
                 httpCode: response.statusCode,
             });
         });
     },
 
     errorResponse: function errorXMLResponse(
-        errCode: ArsenalError,
+        errCode: ArsenalError | Error,
         response: http.ServerResponse,
         log: RequestLogger,
         corsHeaders?: { [key: string]: string } | null,
     ) {
         setCommonResponseHeaders(corsHeaders, response, log);
+        let error;
+        if (errCode instanceof ArsenalError) {
+            error = errCode;
+        } else if (errCode instanceof Error) {
+            error = errors.InternalError.customizeDescription(errCode.message);
```

Comment:
Is there a reason on why this check is inconsistent between the different functions of this file ?

Reply:
I'm not sure I understand fully the question, as we also have this logic in the other `errorResponse` for the json mode, but we work with `err.code` in the functions, something that we do not have if the error is not an Arsenal (or S3 standard) error. In this case, we want to recreate an internal error, so we have a logical error code, and we do not alter the implementation, while allowing any kind of error. Because we cannot guarantee that the error we receive will always be an Arsenal error (the implementation is modular and we use projects like hdclient that do not use Arsenal error, yet hdclient errors can reach this function).